### ISSUE TITLE
Issue with AMD Cape Verde (7700 series) - bluescreens

### ISSUE THREAD

**Issue by fuzzfuzz** [nicehash/NiceHashMiner-Archived#234]
Hi NiceHash!
I need to report an issue which I've had for quite some time, with the mining using NiceHash software on any AMD Radeon 7700 series based card with the Cape Verde chip.
For some time I thought it was based on issues with the AMD drivers, but I've since now replicated this on a second system, which also has a Radeon 7700. Basically, NiceHash starts, correctly detects the card, but as soon as benchmarking of the GPU starts, the system bluescreeens with a Stop 3B System Service Exception error.
My testing so far has included:
On the first system - clearing all drivers, trying different versions of Catalyst, and trying different versions of NiceHash. All the same and all repeatable. I used Guru3d DDU to remove software.
My second system is a new install with Win7 and a Radeon 7770 GHz edition, the issue is present straight away even though it is a new install.
Turning off UAC on both systems.

Digging deeper - I can execute sgminer 5.3 on these systems and it works with no issues. I experimented with replacing the newer sgminer versions in the NiceHash software with this 5.3 version and it does seem to work, to a point, although benchmarking doesn't. So it seems the issue is with a particular sgminer bundled with NiceHash - and it isn't compatible with the Cape Verde.

I have a third system with a Bonaire-based Radeon 7790 and it is fine, no issues whatsoever.

I realise that Cape Verde's are quite old and not the fastest cards in the world but it would be great to get this solved so that I can use NiceHash on all my systems. I can help out with testing if needed.

Thanks! Randall

**Comment by S74nk0** [comment]
Hello @fuzzfuzz. Please tell us what NHM versions were used and which sgminers have been tested manually and/or replaced with the NHM. How long has this been an issue?

**Comment by fuzzfuzz** [comment]
Hi!
It's been an issue with NHM for quite some time - I can't be certain but I believe right back to 1.2 and 1.3.
The version of sgminer currently being used manually for mining is 5.3. This is not present in the newer NHM versions, so again it must have been quite some time ago as NHM is where I sourced it from.
I have not tested newer sgminer versions in the newer NHM. However leave this with me, I will manually test each version and post results.

Thanks.

**Comment by fuzzfuzz** [comment]
Hi,
I've tested for hours this morning.
It occurs all the way back to 1.2.0.0, but I can't pinpoint the exact version of NHM which has the version of sgminer that works. Admittedly I haven't tested every version of NHM, but I did test each of the major versions.
All I can tell you is that the sgminer version is 5.3.0-nicehash-37. This one works. At some point I had transferred this sgminer from the bin folder of NHM, and am using this manually.

How does NHM choose which version of sgminer to use? and for different models of GPU? maybe 7700's need something different? ...

Thanks.
